lib/puppet/provider/cpdb_update/cpdb_update.rb

```diff
@@ -0,0 +1,46 @@
+Puppet::Type.type(:cpdb_update).provide(:cpdb_update) do
+
+  commands :cpdb => '/usr/share/candlepin/cpdb'
+  commands :rpm => 'rpm'
+
+  def create
+    migrate_database
+    update_version_file
+  end
+
+  def exists?
+    Gem::Version.new(previous_candlepin_version) == Gem::Version.new(candlepin_rpm_version)
+  end
+
+  private
+
+  def migrate_database
+    output = cpdb(
+      "--update",
+      "--dbhost=#{resource[:db_host]}",
+      "--dbport=#{resource[:db_port]}",
+      "--database=#{resource[:db_name]}#{resource[:ssl_options]}",
+      "--user=#{resource[:db_user]}",
+      "--password=#{resource[:db_password]}"
+    )
+  end
+
+  def version_file
+    "/var/lib/candlepin/version"
```

Comment:
Perhaps this should be a hidden file?
```suggestion
    "/var/lib/candlepin/.puppet-candlepin-rpm-version"
```